gstaichi/codegen/cpu/codegen_cpu.cpp

```diff
@@ -262,76 +267,85 @@ void KernelCodeGenCPU::optimize_module(llvm::Module *module) {
     options.NoInfsFPMath = 0;
     options.NoNaNsFPMath = 0;
   }
+
   options.HonorSignDependentRoundingFPMathOption = false;
   options.NoZerosInBSS = false;
   options.GuaranteedTailCallOpt = false;
 
-  llvm::legacy::FunctionPassManager function_pass_manager(module);
-  llvm::legacy::PassManager module_pass_manager;
-
+#if LLVM_VERSION_MAJOR >= 18
+  const auto opt_level = llvm::CodeGenOptLevel::Aggressive;
+#else
+  const auto opt_level = llvm::CodeGenOpt::Aggressive;
+#endif
   llvm::StringRef mcpu = llvm::sys::getHostCPUName();
   std::unique_ptr<llvm::TargetMachine> target_machine(
       target->createTargetMachine(triple.str(), mcpu.str(), "", options,
                                   llvm::Reloc::PIC_, llvm::CodeModel::Small,
-                                  llvm::CodeGenOpt::Aggressive));
```

Comment:
ditto for adding things really. If we are going to change things that affect performance, I think we need to run regression tests for before/after the changes.

On the whole, I'd prefer to change as little as possible please. The minimum necessary to get llvm20 working.

We can have a separate PR for performance tweaks, and run various performance benchmarks as part of that PR potentially.